### DISCUSSION TITLE
Multiple Instances of Transmission?

### DISCUSSION THREAD

**Discussion by joe-eklund** [haugene/docker-transmission-openvpn#2628] [Q&A] [answered]
Greetings!

I am attempting to deploy a second instance of this docker stack (both the transmission container and the proxy) on the same machine. I am running into an issue where my second proxy cannot talk to my second transmission instance. 

I believe this is because the nginx config is hardcoded to look for the container name `transmission`. In my second stack, I have changed my transmission container name to `transmission-long`, so it makes sense that the second proxy cannot see the second transmission instance (since it is looking for the container name `transmission` and is not in the same docker network as the first stack).

Error logs:

```
2023-06-03T22:28:57.849621928Z 2023/06/03 15:28:57 [emerg] 1#1: host not found in upstream "transmission" in /etc/nginx/nginx.conf:14
2023-06-03T22:28:57.849676291Z nginx: [emerg] host not found in upstream "transmission" in /etc/nginx/nginx.conf:14
```

Does anyone know if there is any way to customize the transmission container name through something like an environment variable?

My current idea is to have the proxy save the config through a docker mount, then customize the nginx file to my transmission container name, but this feels slightly hacky. And then I have to account for any future image changes and roll those into my custom nginx file too. 

Thoughts?

**Comment by pkishino** [comment, marked as answer]
Are you talking about the webproxy? 
the nginx conf is just the basic setup, as you suggested, mount it and create your own would be best
